### ISSUE TITLE
Trying to show a popup window in an EditorPlugin causes editor to crash

### ISSUE THREAD

**Issue by blackears** [godotengine/godot#78900]
### Godot version

v4.0.3.stable.official [5222a99f5]

### System information

Windows 10, Compatability

### Issue description

I'm trying to open a popup inside of an EditorPlugin and it keeps crashing the editor.  It seems  to be due to the following lines of code:
```
	var base:Node = get_editor_interface().get_base_control()
	base.add_child(popup)
```

### Steps to reproduce

Open the attached project and enable the addon.  Select Project/Tools/Show Test Popup.  The editor will crash.

### Minimal reproduction project

[EditorPopup.zip](https://github.com/godotengine/godot/files/11924987/EditorPopup.zip)

**Comment by AThousandShips** [comment]
I'm not sure how safe it is to directly add nodes to the interface like this

Have you tried "popup_dialog" since you seem to want a popup?

**Comment by blackears** [comment]
No.  Where can I find that?  I don't want to simply use an AcceptDialog or a ConfirmationDialog, because my window is going to be a lot more complex than that.  And the editor only offers Popup and PopupPanel to be the base of a custom control.

**Comment by blackears** [comment]
Also, if there is something other than base_control I could add my popup to, maybe that would help.

**Comment by AThousandShips** [comment]
Under editor interface, though realised that's only available in 4.1

**Comment by quinnyo** [comment]
The test project worked fine for me on swaywm. Have you tried single window mode?

I don't know if it's a good idea or not (or if it will help) but EditorPlugin is a Node and so you can add the Popup directly to it, which might make more sense than the base control, assuming your plugin outlives its popup.

```
Godot Engine v4.0.3.stable.arch_linux
OpenGL API 4.6 (Core Profile) Mesa 23.1.2 - Compatibility - Using Device: AMD - AMD Radeon RX 5600 XT (navi10, LLVM 15.0.7, DRM 3.52, 6.3.9-zen1-1-zen)
```

**Comment by blackears** [comment]
I'm getting a crash when I add the popup directly to the EditorPlugin too.  What is swaywm?  What do you mean by single window mode?

My machine is unable to run Vulkan and I presume is using your DirectX renderer.  Could that have something to do with it?

**Comment by quinnyo** [comment]
> My machine is unable to run Vulkan

That is becoming a bit unusual, it might be helpful if you could share details of your machine.

> single window mode?

In the editor settings you can enable "Single Window Mode" (`interface/editor/single_window_mode`) which will make the popup appear within the main editor (but still as a modal dialog).

Your project was set to use the 'compatibility' renderer which uses OpenGL.
sway is a Wayland compositor / window manager on Linux.

**Comment by blackears** [comment]
Using single window mode did let me see the popup.  I'm concerned, though, that most users won't know to set this option in their version of Godot.

I'm using Windows 10 Pro
AMD FirePro W2100

Let me know if there is other system info you need.

**Comment by blackears** [comment]
Using Mobile without Single Window Mode  set also works.  I do want to be able to compile for web, though.  Can I do that in Mobile mode?

**Comment by TheColorRed** [comment]
When using `popup_dialog`, the editor crashes for me, here is the script I am using:

```gdscript
@tool
extends EditorPlugin

var window

func _enter_tree() -> void:
	window = preload("res://addons/scenes/scanner.tscn").instantiate()
	add_tool_menu_item("Open Window", _open_popup)


func _exit_tree() -> void:
	remove_tool_menu_item("Open Window")


func _open_popup() -> void:
	get_editor_interface().popup_dialog(window)
```

Godot v4.1.2.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated AMD Radeon RX 6700 XT (Advanced Micro Devices, Inc.; 31.0.14037.17019) - AMD Ryzen 7 2700 Eight-Core Processor (16 Threads)

**Comment by ManasMakde** [comment]
Just incase anyone was wondering how to use [`popup_dialog`](https://docs.godotengine.org/en/stable/classes/class_editorinterface.html#class-editorinterface-method-popup-dialog) as mentioned in [this comment](https://github.com/godotengine/godot/issues/78900#issuecomment-1615607321),   
Here's an example in [`@tool mode`](https://docs.godotengine.org/en/stable/tutorials/plugins/running_code_in_the_editor.html)

```gdscript
@tool
extends Node2D

@export var btn := false : set=set_btn

func set_btn(new_val):
	if !new_val:
		return
	
	# create popup
	var label := Label.new()
	label.text = "Loading..."
	label.horizontal_alignment = HORIZONTAL_ALIGNMENT_CENTER
	label.vertical_alignment = VERTICAL_ALIGNMENT_CENTER
	label.set_anchors_preset(Control.PRESET_FULL_RECT)
	var popup := Window.new()
	popup.initial_position = Window.WINDOW_INITIAL_POSITION_CENTER_PRIMARY_SCREEN
	popup.size = Vector2i(150, 50)
	popup.exclusive = true
	popup.borderless = true
	popup.unfocusable = true
	popup.add_child(label)
	
	EditorInterface.popup_dialog(popup)
	
	await get_tree().create_timer(5.0).timeout
	
	popup.queue_free()
```
![Editor Screenshot](https://github.com/user-attachments/assets/d668092f-3649-457c-b236-07f567e5993e)
